### PR TITLE
Runtime notifications permission

### LINKED ISSUE #17714: Runtime permission for notifications
Android 13 users encounter a permission dialog at the first launch of the app. We should control when we show the permission dialog for better UX. 

https://developer.android.com/about/versions/13/changes/notification-permission
Internal P2 post: pbArwn-5EG-p2

### PR DESCRIPTION
**Pull request by irfano**
Fixes #17714 
This PR adds runtime notifications permission support. 

**Notifications screen**
This adds a yellow warning box to the notifications screen. It's visible when the notification permission is not allowed.
- If you tap the "x" button, it's dismissed, and you can't see it unless you allow the permission and disable it again.
- FIX  button opens the system's permission dialog. If you deny the permission several times, the FIX button opens the permission bottom sheet that navigates you to the device settings.
  

**Blogging reminders**
This adds a notification permission step to the blogging reminders flow. If the permission is denied and the app can't show the system's native permission dialog, the "Go to Settings → Notifications → App Settings, and turn WordPress on to be notified immediately." message will be displayed. Otherwise, it will be invisible.
Screenshot 2023-04-07 at 01 33 11

Design: YSbDG4YKoDDkjEZSfBrIlN-fi-1300_13590

To test:
Test both on the API 33 device and a lower version device. The behavior shouldn't change on older devices.

**Notifications screen - Allow**
1. Launch the JP app. 
2. Ensure you don't allow the notification permission.
3. Navigate to Notifications.
4. Notice the yellow warning box.
5. Tap Fix.
6. Ensure it opens the system's permission dialog.
7. Allow permission. 
8. Ensure the warning has been lost and the permission is allowed on system settings.

**Notifications screen - Deny**
1. Launch the JP app. 
2. Ensure you don't allow the notification permission.
3. Navigate to Notifications.
4. Notice the yellow warning box.
5. Tap Fix and deny the native permission dialog.
6. Tap Fix again.
7. Notice the permission bottom sheet appears. 
8. Tap the green button. Notice it sends you to the permission settings screen.

**Blogging reminders screen - Allow**
1. Launch the JP app. 
2. Ensure that notification permission is not allowed.
3. Navigate to My Site → Site Settings.
4. Tap Reminders.
5. Choose some days and tap Update.
6. Notice the native permission dialog.
7. Allow it and test the positive case.

**Blogging reminders screen - Deny**
1. Launch the JP app. 
2. Ensure that notification permission is not allowed.
3. Navigate to My Site → Site Settings.
4. Tap Reminders.
5. Choose some days and tap Update.
6. Deny the native permission dialog.
7. It should open permission bottom sheet.
8. Tap the "Turn on notifications" button.
9. Ensure it navigates you to the permission settings screen.
10. Allow the notification permission and go back to the app.
11. Ensure the permission screen is passed and the next screen on the flow is opened.

## Regression Notes
1. Potential unintended areas of impact
Setting blogging Reminders.

12. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually.

13. What automated tests I added (or what prevented me from doing so)
Updated current tests.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ravishanker** [comment]
@irfano - Great work!

- **Notifications Screen - Allow**: Suggest that entire banner is made clickable.  Fix word and touch target is obscure.  Not sure if it needs that x.  When banner is dismissed it disappears, they may not know how to enable notifications just in case!
- **Notifications Screen - Deny**: Didn't have to deny several time to see the Bottom sheet

|Banner|Bottomsheet|Popup|
|-|-|-|
| Screenshot_20230406_174700|Screenshot_20230406_174952| Screenshot_20230406_175142|

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1497,6 +1497,15 @@
     <string name="notifications_tab_title_follows">Follows</string>
     <string name="notifications_tab_title_likes">Likes</string>
 
+    <!-- Notification permission -->
+    <string name="notifications_permission_fix">Fix</string>
+    <string name="notifications_permission_dismiss_content_description">Dismiss notification permission warning.</string>
+    <string name="notifications_permission_off_desc">Push notifications are turned off.</string>
+    <string name="notifications_permission_bottom_sheet_title">Push notifications are turned off</string>
+    <string name="notifications_permission_bottom_sheet_description_1">You’ll need to open the app to see notifications.</string>
```
Comment (ravishanker) [inline WordPress/src/main/res/values/strings.xml]: Missing escape \ character before ' in You'll.  Also, same below L:4071

**Review by ravishanker — APPROVED** [review]
Finally, it's done! 🚀

**Comment by irfano** [comment]
> Suggest that entire banner is made clickable. Fix word and touch target is obscure.

That makes sense 👍🏻 ~I'll make the change.~ Done: 63431aaae797679b35f73e0b0a50648e608680fb

> Not sure if it needs that x. When banner is dismissed it disappears, they may not know how to enable notifications just in case!

Some users may be aware that notifications are turned off and just want to close the banner. But as a side effect, other users who are unaware of it may tap "X" by mistake and lose the benefit of the banner. 
Maybe we can remove the "X" button and hide the banner if permission is always denied. Wdyt?

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (irfano) [reply WordPress/src/main/res/values/strings.xml]: I've learned that `’` and `'` are different. 😄 I copied strings from Figma, which uses `’`, that's why the IDEA didn't show an error. But I've changed `’` to `'`. They both are okay but standard `'` is more popular in our strings.

Done: 6ee77da96b346ea0569b234c9d1f21c96168c19d

**Comment by irfano** [comment]
> Didn't have to deny several time to see the Bottom sheet

I think it changes that how many times you can deny the system's native permission dialog until we can't show it anymore. Because I can see and deny the system's native permission dialog twice after the clean install. 
I've also made a change to the blogging reminders flow. af470adfb3c0c77d65a714f06784f89ace026822 See the updated description.

**Pull request merged**